changelog-extractor.py

```diff
@@ -0,0 +1,144 @@
+                if issueType not in mappedIssues :
+                    mappedIssues.update({issueType : [line]})
+                else:
+                    mappedIssues[issueType].append(line)
+                break
+        if categorizedIssue :
+            continue
+        #if code reach that line - means issue type is not in issueTypes -> typo or uncategorized issuetype
+        if uncategorizedIssueType not in mappedIssues :
+            mappedIssues.update({uncategorizedIssueType : [line]})
+        else:
+            mappedIssues[uncategorizedIssueType].append(line)
+        continue      
+    return mappedIssues
+
+lineBreaker = "\n"
+notEnoughInputParametersErr = "Not enough input parameters"
+issueTypes = {"Enhancement","Fix", "Feature", "Ongoing", "Checkmark",
+"Related", "Lab", "Live", "Refactor", "Nochangelog", "Technical"}
+
+uncategorizedIssueType = "Uncategorized"
+
+def ConvertChangelogTextToMd(nonFormattedText , header ) : 
+    """Returns .MD formatted changelog based on raw formatted text.
+     Header - 'title' for set of issues in that changelog
+    """
+    mappedIssues = parseRawChangelog(nonFormattedText)
+    if len(mappedIssues) == 0 :
+        return ""
+    res = ""
+    if not (not header or header == ""):
+      res += buildHeaderProject(header) + lineBreaker
+    res += buildChangelogBody(mappedIssues)
+    return res
+
+def buildHeaderProject(header )  :
+	return "## " + header + lineBreaker
+
+def buildHeaderIssue(header )  :
+	return "### " + header + ":" + lineBreaker
+
+def buildIssue(issue ) :
+	return " - " + issue
+
+def buildChangelogBody(mappedIssues)  :
+	res = ""
+	for issueType  in  mappedIssues :
+		res += buildHeaderIssue(issueType)
+		for issue in mappedIssues[issueType] :
+			res += buildIssue(issue) + lineBreaker
+		res += lineBreaker
+	return res
+
+def buildMessageToSlack(tag) :
+    return os.environ['CI_PROJECT_URL']  + "/tags/" + tag
+
+def sendMessageToSlack(text, channel, token, username) :
+    """Sends message to slack
+        text - text to be send
+        channel - channel name at slack. example: #general
+        token - bot token for posting messages
+        username - bot name
+    """
+    #http post request to slack service to post message to specific channel
+    command = 'curl https://slack.com/api/chat.postMessage -X POST -d "as_user=false" -d "channel='+ channel+'" -d "username='+username+'" -d "token='+token+'" -d "text='+text + '"'
+    os.system(command)
+
+def main():
+    newCommit = os.environ['CI_COMMIT_SHA']
+    oldCommit = ""
+    if "CI_BUILD_TAG" in os.environ:
+        newTag = os.environ["CI_BUILD_TAG"]
+        oldCommit = getCommitByTag(sendCommand("git describe --abbrev=0 --tags "+ newTag +"^"))
+    else :
+        oldCommit = getCommitByTag(sendCommand("git describe --abbrev=0 --tags"))
+
+    mdFormattedChangelog = getMdFormattedChangelog(newCommit, oldCommit)
+
+    print( mdFormattedChangelog)
+
+    if "CI_BUILD_TAG" in os.environ:
```

Comment:
Declare newTag="" at the beggining, and check here if the tag was setted.
We can avoid having two references for CI_BUILD_TAG